server/contexts/ctxerr/ctxerr.go

```diff
@@ -0,0 +1,78 @@
+// Package ctxerr provides functions to wrap errors with annotations and
+// stack traces, and to handle those errors such that unique instances of
+// those errors will be stored for an amount of time so that it can be
+// used to troubleshoot issues.
+//
+// Typical uses of this package should be to call New or Wrap[f] as close as
+// possible from where the error is encountered (or where it needs to be
+// created for New), and then to call Handle with the error only once, after it
+// bubbled back to the top of the call stack (e.g. in the HTTP handler, or in
+// the CLI command, etc.). It is fine to wrap the error with more annotations
+// along the way, by calling Wrap[f].
+package ctxerr
+
+import (
+	"context"
+	"time"
+
+	"github.com/fleetdm/fleet/v4/server/errorstore"
+	"github.com/pkg/errors"
+	"github.com/rotisserie/eris"
+)
+
+type key int
+
+const errHandlerKey key = 0
+
+// NewContext returns a context derived from ctx that contains the provided
+// error handler.
+func NewContext(ctx context.Context, eh *errorstore.Handler) context.Context {
+	return context.WithValue(ctx, errHandlerKey, eh)
+}
+
+func fromContext(ctx context.Context) *errorstore.Handler {
+	v, _ := ctx.Value(errHandlerKey).(*errorstore.Handler)
+	return v
+}
+
+// New creates a new error with the provided error message.
```

Comment:
Maybe also add a `Errorf`?

Reply:
Yeah maybe, I'll add it if needed in the subsequent PR that will start using this.